recipes/libdeflate/all/conanfile.py

```diff
@@ -60,51 +56,65 @@ def configure(self):
                 self.tool_requires("msys2/cci.latest")
 
     def source(self):
-        get(self, **self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        env = VirtualBuildEnv(self)
+        env.generate()
+
+        if is_msvc(self) or self._is_clangcl:
+            vc = VCVars(self)
+            vc.generate()
+        else:
+            tc = AutotoolsToolchain(self)
+            tc.generate()
 
-    def _build_msvc(self):
-        with chdir(self, self._source_subfolder):
-            with vcvars(self), environment_append(VisualStudioBuildEnvironment(self).vars):
-                target = "libdeflate.dll" if self.options.shared else "libdeflatestatic.lib"
-                self.run("nmake /f Makefile.msc {}".format(target))
+    def _build_nmake(self):
+        with chdir(self, self.source_folder):
+            target = "libdeflate.dll" if self.options.shared else "libdeflatestatic.lib"
+            self.run(f"nmake /f Makefile.msc {target}")
 
     def _build_make(self):
-        autotools = AutoToolsBuildEnvironment(self, win_bash=(self._settings_build.os == "Windows"))
-        with chdir(self, self._source_subfolder):
+        autotools = Autotools(self)
+        with chdir(self, self.source_folder):
             autotools.make()
 
     def build(self):
         apply_conandata_patches(self)
         if is_msvc(self) or self._is_clangcl:
-            self._build_msvc()
+            self._build_nmake()
         else:
             self._build_make()
 
     def _package_windows(self):
-        self.copy("libdeflate.h", dst="include", src=self._source_subfolder)
+        copy(self, "libdeflate.h", dst=os.path.join(self.package_folder, "include"), src=self.source_folder)
         if self.options.shared:
-            self.copy("*deflate.lib", dst="lib", src=self._source_subfolder)
-            self.copy("*deflate.dll", dst="bin", src=self._source_subfolder)
+            copy(self, "*deflate.lib", dst=os.path.join(self.package_folder, "lib"), src=self.source_folder)
+            copy(self, "*deflate.dll", dst=os.path.join(self.package_folder, "bin"), src=self.source_folder)
         else:
-            self.copy("*deflatestatic.lib", dst="lib", src=self._source_subfolder)
+            copy(self, "*deflatestatic.lib", dst=os.path.join(self.package_folder, "lib"), src=self.source_folder)
 
     def _package_make(self):
-        autotools = AutoToolsBuildEnvironment(self, win_bash=(self._settings_build.os == "Windows"))
-        with chdir(self, self._source_subfolder):
-            autotools.install(args=["PREFIX={}".format(self.package_folder)])
+        autotools = Autotools(self)
+        with chdir(self, self.source_folder):
+            autotools.install(args=[f"PREFIX={self.package_folder}"])
```

Comment:
Should this use `DESTDIR` like the [Autotools template](https://github.com/conan-io/conan-center-index/blob/23c06c9e751e6fc29ad655335873378cf75d9e7e/docs/package_templates/autotools_package/all/conanfile.py#L170-L171)?

```suggestion
            # TODO: replace by autotools.install() once https://github.com/conan-io/conan/issues/12153 fixed
            autotools.install(args=[f"DESTDIR={unix_path(self, self.package_folder)}"])
```

Reply:
Thanks!

Reply:
@jwillikers @uilianries this didn't actually work on linux, see comment below.